### PR TITLE
feat: log initial connection to the network, printing if client

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

**Review by RolandSherwin — APPROVED** [review]
Printing the commit hash would prevent some confusions, atleast for me!

File: safenode/src/client/api.rs [hunk 1]
```diff
@@ -91,23 +92,31 @@ impl Client {
             match event {
                 ClientEvent::ConnectedToNetwork => {
                     info!("Client connected to the Network.");
+                    println!("Client successfully connected to the Network.");
```
Comment (maqi) [inline safenode/src/client/api.rs:94]: user will only see this `println` , so actually only need to check the `logged_connected` flag here?
then don't need to pass the flag down and do other changes?

File: safenode/build.rs [hunk 2]
```diff
@@ -5,9 +5,26 @@
...
+        .args(["rev-parse", "HEAD"])
+        .output()
+        .expect("Failed to execute git command");
+
+    let git_hash = String::from_utf8(output.stdout).unwrap().trim().to_string();
+
+    let mut file = File::create("src/git_hash.rs").expect("Failed to create git_hash.rs");
+    writeln!(
+        file,
+        "/// The git commit at build time\npub const GIT_HASH: &str = \"{}\";",
```
Comment (maqi) [inline safenode/build.rs]: shall the detailed `build time` (i.e. `Instant::now()` in readable format) got written as well ?

Comment (joshuef) [reply safenode/build.rs]: I guess that could be added if you think it's useful?

File: safenode/src/client/api.rs (hunk 1 above)
Comment (joshuef) [reply safenode/src/client/api.rs:94]: Ah yeh, clients already did it only once

File: safenode/build.rs (hunk 2 above)
Comment (joshuef) [reply safenode/build.rs]: I'm not sure it is? Does it matter if we have the commit? Gonna leave that off, we can always add it in later if desired 👍

**Pull request merged**